### ISSUE TITLE
Location files loaded twice and never cached if access-control-allow-origin header is set to single URL

### ISSUE THREAD

**Issue by estruyf** [SharePoint/sp-dev-docs#6193]
## Category

- [x] Bug

## Describe the bug

We noticed a strange behavior in the files which get loaded for SPFx web parts and extensions. Each localization file is requested twice. First from the page bundle (default SharePoint and fetches it with cors) and second from requirejs (fetches it with no-cors).

![Screenshot 2020-08-21 at 13 26 32](https://user-images.githubusercontent.com/2900833/90885750-f3f72b80-e3b1-11ea-97e2-6a0fcc5ce2f4.png)

Because we wanted to see what was happening here, we noticed another issue. The localization files are not cached when you use your own CDN like Azure CDN for example and the `access-control-allow-origin` is set to the origin of your tenant `https://<tenant>.sharepoint.com`, but when providing a wildcard. It will cache the files correctly.

**With the single URL reference**

![Screenshot 2020-08-21 at 11 54 24](https://user-images.githubusercontent.com/2900833/90886431-a9c27a00-e3b2-11ea-9e30-4adb18a1d741.png)

The file is allowed to get fetched as the tenant origin is present in the `access-control-allow-origin`, but this is never cached.

I also tested it outside of SharePoint from a simple website with the same headers (except for the domain) and the file gets cached correctly.

![Screenshot 2020-08-21 at 13 34 54](https://user-images.githubusercontent.com/2900833/90886685-27868580-e3b3-11ea-85c2-bf76ca52d7d3.png)

![Screenshot 2020-08-21 at 13 35 06](https://user-images.githubusercontent.com/2900833/90886693-2bb2a300-e3b3-11ea-9ff2-2c3ab65d95b0.png)

**With * wildcard**

![Screenshot 2020-08-21 at 13 28 23](https://user-images.githubusercontent.com/2900833/90886718-3b31ec00-e3b3-11ea-8ede-d4fcba3be59d.png)

File gets still requested twice, but it will get cached correctly.

## Steps to reproduce

- Create a new web part
- Create an Azure CDN or you can also test it from table storage to be quicker
- Set the CORS settings on the table storage
- Bundle and package your project
- Upload it to SharePoint and check the localization files which are loaded

## Expected behavior

Expected behaviour: should only load the localization file once and should cache it correctly even if no wildcard CORS setting is used.

## Environment details (*development & target environment*)

> **DELETE THIS LINE BEFORE SUBMITTING** - Complete the following (*if ignored you'll be prompted for it before we can address your issue... save the time and provide it in your initial submission*).

- **Your Developer Environment**: both Windows and macOS
- **Target Environment**: SharePoint Online
- **Browser(s)**: All Chromium-based browsers experience this issue. Not happening on Firefox.
